File: frontend/src/app/models/Article.ts
import { Comment } from './Comment';

export interface Article {
  uuid?: string;
  userId?: string;
  username?: string;
  title: string;
  body: string;
  tags: string[];
  images?: File[];
  comments?: Comment[];
  totalPoints?: number;
  userLike?: number;
  createdAt?: Date;
  updatedAt?: Date;
}

File: new-backend/Models/Article.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Newtonsoft.Json;
namespace Makro.Models
{
    public class Article
    {
        [JsonIgnore]
        public int Id { get; set; }
        [Required]
        public string UUID { get; set; }
        [Required]
        public User User { get; set; }
        [Required]
        public string Title { get; set; }
        [Required]
        public string Body { get; set; }
        public ICollection<ArticleImage> Images { get; set; }
        public ICollection<Comment> Comments { get; set; }
        [Required]
        public List<string> Tags { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
    }
}
